include/picongpu/fields/MaxwellSolver/CKC/CKC.hpp

```diff
@@ -0,0 +1,90 @@
+/* Copyright 2013-2023 Axel Huebl, Heiko Burau, Rene Widera, Remi Lehe,
+ *                     Sergei Bastrakov, Lennert Sprenger
+ *
+ * This file is part of PIConGPU.
+ *
+ * PIConGPU is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PIConGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "picongpu/simulation_defines.hpp"
+
+#include "picongpu/fields/MaxwellSolver/CFLChecker.hpp"
+#include "picongpu/fields/MaxwellSolver/CKC/CKC.def"
+#include "picongpu/fields/MaxwellSolver/CKC/Derivative.hpp"
+#include "picongpu/fields/MaxwellSolver/DispersionRelation.hpp"
+#include "picongpu/fields/MaxwellSolver/FDTD/FDTD.hpp"
+
+#include <pmacc/algorithms/math/defines/pi.hpp>
+#include <pmacc/traits/GetStringProperties.hpp>
+
+
+namespace picongpu
+{
+    namespace fields
+    {
+        namespace maxwellSolver
+        {
+            /** Specialization of the CFL condition checker for CKC solver
+             *
+             * @tparam T_Defer technical parameter to defer evaluation
+             */
+            template<typename T_Defer>
+            struct CFLChecker<CKC, T_Defer>
+            {
+                /** Check the CFL condition according to the paper, doesn't compile when failed
+                 *
+                 * @return value of 'X' to fulfill the condition 'c * dt <= X'
+                 */
+                float_X operator()() const
+                {
+                    // cellSize is not constexpr currently, so make an own constexpr array
+                    constexpr float_X step[3] = {CELL_WIDTH, CELL_HEIGHT, CELL_DEPTH};
```

Comment:
It was called `step` because it how much you step in a direction when you go one cell and because it is oriented on the [Lehe solver](https://github.com/ComputationalRadiationPhysics/picongpu/blob/5456d097efc3c6e60bb395e11ff452a3b280963e/include/picongpu/fields/MaxwellSolver/Lehe/Derivative.hpp) implementation

Reply:
Thanks! I still don't think that this is a very intuitive name but if that's the jargon of this domain, that's fine with me.